### PR TITLE
Use order key in query header if defined

### LINKED ISSUE #148: [FEATURE] Order widgets using `order` key in query header
Order the widgets using the [`order` key](https://github.com/databrickslabs/lsql/blob/main/docs/dashboards.md#widget-ordering) stated in the [query header](https://github.com/databrickslabs/lsql/blob/main/docs/dashboards.md#headers-of-sql-files).

### PR DESCRIPTION
**Pull request by JCZuurmond**
Resolves #148 

Screenshot 2024-06-12 at 14 47 22

### CODE REVIEW

**Comment by JCZuurmond** [comment]
PR is stacked upon #147

File: src/databricks/labs/lsql/dashboards.py [hunk 1]
```diff
@@ -45,6 +49,38 @@ def as_dict(self) -> dict[str, str]:
         return dataclasses.asdict(self)
 
 
+@dataclass
+class WidgetMetadata:
+    order: int
+    width: int
+    height: int
```
Comment (nfx) [inline src/databricks/labs/lsql/dashboards.py:56]: if widgets are optional, they have to get a default value. or `| None = None`

**Review by nfx — APPROVED** [review]
lgtm

File: src/databricks/labs/lsql/dashboards.py (hunk 1 above)
Comment (JCZuurmond) [reply src/databricks/labs/lsql/dashboards.py:56]: Yes, currently it is not required yet as the default options are provided when creating the fallback metadata

**Pull request merged**